File: packages/web/src/features/student/studentPresentation.test.ts
import { describe, expect, it } from 'vitest';
import { isEarlyReaderPresentation } from './studentPresentation';

describe('isEarlyReaderPresentation', () => {
  it('is true for Pre-K/K and grades 1-2', () => {
    expect(isEarlyReaderPresentation({ gradeLevel: 'K' })).toBe(true);
    expect(isEarlyReaderPresentation({ gradeLevel: '1' })).toBe(true);
    expect(isEarlyReaderPresentation({ gradeLevel: '2' })).toBe(true);
  });

  it('is false for grades 3 and up', () => {
    expect(isEarlyReaderPresentation({ gradeLevel: '3' })).toBe(false);
    expect(isEarlyReaderPresentation({ gradeLevel: '6' })).toBe(false);
  });

  it('fails closed to the standard (non-simplified) presentation when the grade level is unresolvable', () => {
    expect(isEarlyReaderPresentation({ gradeLevel: undefined })).toBe(false);
    expect(isEarlyReaderPresentation({ gradeLevel: '9' })).toBe(false);
  });
});

File: packages/mobile/test/features/student/student_presentation_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:sprout/core/models/student.dart';
import 'package:sprout/features/student/student_presentation.dart';

Student _student({String? gradeLevel}) => Student(
  id: 's1',
  firstName: 'Alex',
  lastName: 'Rivera',
  displayName: 'Alex Rivera',
  balanceCents: 0,
  ownerUids: const [],
  gradeLevel: gradeLevel,
);

void main() {
  test('is true for Pre-K/K and grades 1-2', () {
    expect(isEarlyReaderPresentation(_student(gradeLevel: 'K')), isTrue);
    expect(isEarlyReaderPresentation(_student(gradeLevel: '1')), isTrue);
    expect(isEarlyReaderPresentation(_student(gradeLevel: '2')), isTrue);
  });

  test('is false for grade 3 and up', () {
    expect(isEarlyReaderPresentation(_student(gradeLevel: '3')), isFalse);
    expect(isEarlyReaderPresentation(_student(gradeLevel: '6')), isFalse);
  });

  test('fails closed to the standard presentation when unresolvable', () {
    expect(isEarlyReaderPresentation(_student()), isFalse);
    expect(isEarlyReaderPresentation(_student(gradeLevel: '9')), isFalse);
  });
}
